### PR TITLE
custom_rules: Allow a line to be excluded multiple times.

### PR DESCRIPTION
**Pull request by amanagr**
For a regex, if an excluded line in present twice in a file, we
should be able to exclude it twice.

Conversation in https://chat.zulip.org/#narrow/stream/6-frontend/topic/realm.20creation.20form

### CODE REVIEW

**Review by andersk — CHANGES_REQUESTED** [review]
We can’t just skip the `exclude_lines.remove`, because later there’s

```python
        if exclude_lines:
            print('Please remove exclusions for file %s: %s' % (fn, exclude_lines))
```

**Comment by amanagr** [comment]
Okay! updated.

**Comment by andersk** [comment]
I mean the `print()` wasn’t there for no reason, it’s there to stop us from accumulating dead exclusions. We should preserve this functionality, just correctly.

**Comment by garg3133** [comment]
I think instead of removing elements from the original set, we could add the matched elements to a new set and apply `difference_update` on both sets at last.

Or even better, we could make a copy of original set at the start and instead of removing elements from the original set, `discard` the matched elements from the second set. Then we can use the second set to print the dead exclusions.

**Comment by amanagr** [comment]
Yeah, makes sense. Thanks @andersk and @garg3133. Updated!

File: zulint/custom_rules.py [hunk 1]
```diff
@@ -109,14 +109,15 @@ def check_file_for_pattern(
...
+        unmatched_exclude_lines = exclude_lines.copy()
 
         pattern = re.compile(rule['pattern'])
         strip_rule = rule.get('strip')  # type: Optional[str]
 
         ok = True
         for (i, line, line_newline_stripped, line_fully_stripped) in line_tups:
             if line_fully_stripped in exclude_lines:
-                exclude_lines.remove(line_fully_stripped)
+                unmatched_exclude_lines.remove(line_fully_stripped)
```
Comment (garg3133) [inline zulint/custom_rules.py]: I think `remove` should be replaced with `discard` as `remove` raises an error if the element is not present in the set which will be true while excluding the same line multiple times.

Comment (amanagr) [reply zulint/custom_rules.py]: Cool! Thanks!

**Pull request merged**